policy/engine.go

```diff
@@ -488,3 +493,12 @@ func removeRulePrefix(rule string) string {
 
 	return rule
 }
+
+type printHook struct {
+	s *[]string
+}
+
+func (ph printHook) Print(pctx print.Context, msg string) error {
+	*ph.s = append(*ph.s, fmt.Sprintf("%v: %v\n", pctx.Location, msg))
```

Comment:
super nitpick: `%v` shouldn't be used when we know the input is a string for `msg`